### DISCUSSION TITLE
Aumentar a velocidade da transferência ao pressionar o mouse (hold)?

### DISCUSSION THREAD

**Discussion by dropecosta** [frontendbr/forum#987] [Perguntas e Respostas]
Estou tentando utilizar os botões para transferir o valor de um usuário para o outro, mas gostaria de, ao clicar e pressionar o mouse (hold) aumentar a velocidade, já quando clicar apenas, passar os valores de um a um.

Alguém pode me ajudar? Já comecei a fazer algumas tentativas com a função addSpeed(), no JS, mas ainda sem sucesso.
Segue [JsFIDDLE](https://jsfiddle.net/9xamta0b/5/)

**Comment by Marlysson** [comment]
Só tem dois botões e as mesmas letras de cada lado da view.. O que era pra acontecer quando clicasse/segurasse..

**Comment by dropecosta** [comment]
Está funcionando localmente, realmente a view não mostrou,  era para transferir os valores de um usuário para o outro. Mas a questão é que estou tentando, quando ao clicar e pressionar alterar a velocidade.

**Comment by kvnol** [comment]
Não sei se vou conseguir ajudar, mas tô reproduzindo essa feature no meu Codepen com Vanilla. Se eu conseguir, mando aqui.

**Comment by dropecosta** [comment]
Valeu pela força @kvnol, obrigado!

**Comment by kvnol** [comment]
@dropecosta estou muito feliz por dois motivos: por estar conseguindo te ajudar e por eu ter conseguido resolver esse problema.

Olha, resolvi o problema utilizando os eventos `mousedown` e `mouseup` junto com `setInterval()`. Se liga: https://kvnol.github.io/studiesjs/features/lendValues/

O source está aqui: https://github.com/kvnol/studiesjs/tree/master/features/lendValues

:tada: :tada: :tada:

**Comment by kvnol** [comment]
Conseguiu resolver, @dropecosta?

**Comment by dropecosta** [comment]
Fala @kvnol , desculpa pela demora em responder. Na verdade, consegui implementar o seu código, apesar de bem mais simples e limpo, ele é similar ao meu antigo (no funcionamento), onde somente um valor (no mousedown) regula a velocidade. Seu aumentar, não consigo passar de um em um. O que precisava era que a velocidade aumentasse bastante ao estar com o mouse pressionado, mas quando clicar apenas, ele incrementar de um em um.

Achei uma solução que faz isso, mas de uma forma mais simples. Ele não tem dois valores, só vai para um lado. 
https://codepen.io/maehana/pen/Gxqgpy

Na verdade, estou tentando implementar essa lógica para os dois, mais ainda sem sucesso.

**Comment by dropecosta** [comment]
O código está em:
https://jsfiddle.net/exppt6gp/

**Comment by kvnol** [comment]
> O que precisava era que a velocidade aumentasse bastante ao estar com o mouse pressionado

Mas aí é só alterar o valor de tempo do `setInterval`.

**Comment by dropecosta** [comment]
mas o que acontece é que quando aumenta o valor do tempo, ele não consegue passar mais de um em um, porque é só um valor que regula o tempo.

Tipo, se você só clicar e o tempo do clique for menor do que 100ms (um clique normal) ele conta 1 por 1, se você deixar clicado, quando passar de 100ms ele vai somando 1 por 1 a cada 0ms, ou seja, bem rápido

Mas com esse código não estou conseguindo faz a relação entre os dois valores...

**Comment by felipefialho** [comment]
@dropecosta Quando for responder, na sequencia de um comentário seu, edite a mensagem, se não configura flood e ninguém vai querer ajudar.

**Comment by dropecosta** [comment]
Beleza, desculpe @LFeh, tirei o código da página, coloquei no Fiddle, acho que melhorou.

**Comment by dropecosta** [comment]
Boa tarde pessoal,

Consegui implementar parcialmente a lógica que regula a velocidade como tinha imaginado.

Segue o desenvolvimento. Está em:
https://codepen.io/dropecosta/pen/eMZEez

Tenho agora novas questões para compartilhar e pedir sugestões de impelementação.

1 - Preciso que o valor mínimo seja 0.01. Estou tentando implementar isso através da funcão (comentada) minimalValues, mas não está dando certo.

Preciso que esse valor mínimo realmente pare na 0.01. Atualmente, quando corro com o hold, ele acaba parando em -0.01, -0.03, o que não é legal.

Alguém me dá uma força? 

Obrigado!!!
